### PR TITLE
Fixes #374 Removed deprecated classes and methods

### LINKED ISSUE #374: Remove deprecated API from Mockito 2
#This is a reminder to remove all deprecated API's from the upcoming mockito 2.

This issue blocks #123 [Release Mockito 2.0](123)

Relates to 

-  #273 [Get rid of ReturnValues](273)

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
Fixes #374 removed deprecated classes and methods

### CODE REVIEW

File: src/main/java/org/mockito/junit/MockitoJUnit.java [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
-     * Creates a rule instance that can perform lazy verifications.
-     *
-     * @see VerificationCollector
-     * @return the rule instance
-     */
-    public static VerificationCollector collector() {
-        return new VerificationCollectorImpl();
-    }
-}
+package org.mockito.junit;
```
Comment (szpak) [inline src/main/java/org/mockito/junit/MockitoJUnit.java]: What does this file look like completely rewritten (still having visually many lines intact)?

File: src/main/java/org/mockito/verification/VerificationWithTimeout.java [hunk 2]
```diff
@@ -1,122 +1,88 @@
...
-     * 
-     * @return verification mode
-     */
-    public VerificationMode only();       
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+
+package org.mockito.verification;
```
Comment (szpak) [inline src/main/java/org/mockito/verification/VerificationWithTimeout.java]: The same here.

File: src/main/java/org/mockito/junit/MockitoJUnit.java (hunk 1 above)
Comment (ChristianSchwarz) [inline src/main/java/org/mockito/junit/MockitoJUnit.java]: It seems like eclipse reformatted the whole file (Unix vs Win line ending). I guess it is related to [this eclipse bug](https://bugs.eclipse.org/bugs/show_bug.cgi?id=361503). I will test the workaround described in [this stackoverflow answer](http://stackoverflow.com/a/18696063/932656).

**Comment by raphw** [comment]
Thanks for this PR and for fixing the formatting. Could you also squash all commits to a single one for better versioning? Thanks!

**Comment by ChristianSchwarz** [comment]
@raphw I fixed the line ending issue. 

Is it okay to squash the commits once the review of all commits is fine? The removal of some deprecated classes and methods was more than simply removing it, e.g. MockitoJUnitRule. If my implementation is not okay it can easily be reverted.

**Comment by ChristianSchwarz** [comment]
@raphw Do you know why the check "codecov/changes" fail?

**Comment by stevepeak** [comment]
@ChristianSchwarz fixing issue and will report back shortly.

**Comment by stevepeak** [comment]
@ChristianSchwarz can you submit an empty commit for another rebuild. I believe this issue was fixed within the last 6 days. Thank you

**Comment by raphw** [comment]
I do not currently have good Internet access (vacation). Please repush (forced) and see if it works.

**Comment by ChristianSchwarz** [comment]
@raphw and @stevepeak Thanks for your support

**Comment by ChristianSchwarz** [comment]
@raphw @szczepiq  @bric3 
Sorry for stressing you. The people want Mockito 2 & 3 (Lambda). Please comment and/or merge this PR, its a little step in the right direction.

**Comment by ChristianSchwarz** [comment]
ping....

**Comment by TimvdLippe** [comment]
LGTM, thanks for doing the chore work!

**Pull request merged**